devops/scripts/asa/README.md

```diff
@@ -0,0 +1,39 @@
+Scripts in this directory allows to run 
+[microsoft/AttackSurfaceAnalyzer](https://github.com/microsoft/AttackSurfaceAnalyzer) 
+scans for
+[Azure/azure-osconfig](https://github.com/Azure/azure-osconfig)
+
+Scans are running in the container.
+`podman` or `docker` on host system is needed to run the script. 
+The Default is `docker`, can be configured in [run_asa.sh](run_asa.sh).
+
+## How to run:
+
+`./run_asa.sh -p https://github.com/Azure/azure-osconfig/releases/download/test_policy_package/AzureLinuxBaseline.zip`
+
+Run `./run_asa.sh` without any options to see the possible options.
+
+Script will fail if ASA scan detect any findings with severity equal to error
+(by default, can be configured using `-fo` option)
+
+The `./report` directory will be created and following output files will be put there:
+* ./report/after_audit - ASA report after audit, together with related osconfig_nrp.log
+* ./report/after_remediation - ASA report after remediation, together with related osconfig_nrp.log
+* ./report/asa.log - output log from all of the ASA tool collect and report generation runs
```

Comment:
How will be problems detected by ASA reported and will the CI job fail? Or this ASA job will only run the collection of logs and we'll require manual reading of the collection logs to find problems, if any?

Reply:
It's described just a few lines above in the same file:
```
Script will fail if ASA scan detect any findings with severity equal to error
(by default, can be configured using `-fo` option)
```

It can be configured to fail on eg. warnings and anything above by:
`./run_asa.sh -fo warning -p https://aka.ms/AzureLinuxBaseline.zip`

Reply:
RIght now, after manually running the script, it will give you the summary of all severities after the scan is done, like:
```
[18:30:04 INFO] Checking after audit ASA report...

error: 0

warning: 2
 - SetGid: 1
 - SetUid: 1

note: 574
 - Default Level: 574
Check: exiting with return code 2 due to issues at or above warning severity
```
I will do my best to make the workflow in the way that the person who has no idea what is this pipeline for will be able to easily find the cause of why it fails.

Reply:
And after it you have the full report generated in report directory, which is documented. However, I can add a line to the log like:
`You can find details in the generated reports in the report directory`
if you find it useful. WDYT?

Reply:
No need for this other line in my opinion. Thank you for this work, very useful to our project overall!